### ISSUE TITLE
ExplicitCollectionElementAccessMethod reports put() usages where the return value is used

### ISSUE THREAD

**Issue by Gama11** [detekt/detekt#4122]
The `ExplicitCollectionElementAccessMethod` rule complains about the following pattern:

```kotlin
val map = mutableMapOf<String, String>()
val alreadyPresent = map.put("foo", "bar") != null
```

This seems similar to #2391 in that you can't easily replace this with an indexed access, so it probably shouldn't be reported.

```kotlin
// Compiler error: "Assignments are not expressions, and only expressions are allowed in this context"
val alreadyPresent = (nodes["foo"] = "bar") != null
```
____

_Using detekt 1.18.1 with the gradle plugin._
